### DISCUSSION TITLE
Difference between KuberenetesExecutor and LocalKubernetesExecutor

### DISCUSSION THREAD

**Discussion by llamageddon83** [apache/airflow#31858] [Q&A] [answered]
Hi all,

I need clarification about the difference between the two executors. Does using `LocalKubernetes` mean that the tasks from one DAG will run on the same pod? Like what would be the difference if we ran Airflow on Kubernetes with `KuberenetesExecutor` and switched to `LocalKubernetesExecutor`?

And if my understanding is incorrect, is there a way to run tasks from one DAG on the same pod? 

Thank you

**Comment by potiuk** [comment, marked as answer]
You can choose which tasks are used locally and which on K8S - see the docs of ours also similar CeleryKubernetesExecutor).

If you run several tasks on the same "LocalExecutor" (i.e. without assigning them to kubernetes queue and if you have one Scheduler, then yes, they will be all executed on scheduler (this is where LocalExecutor tasks are executed).

**Reply by llamageddon83** [reply]
Thank you
